### PR TITLE
Create CSP docs for 2.4

### PR DESCRIPTION
**Pull request by AlexMaxHorkun**
New CSP changes for 2.4.0 require additional docs

## Purpose of this pull request

This pull request (PR) document new 2.4-specific changes to CSP support

## Affected DevDocs pages

-  https://devdocs.magento.com/guides/v2.4/extension-dev-guide/security/content-security-policies.html

Added a new section about whitelisting inline scripts and styles in a [content security policy](https://devdocs.magento.com/guides/v2.4/extension-dev-guide/security/content-security-policies.html#whitelist-an-inline-script-or-style).

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+configure policies to work according to their custom code. After the policies have been configured,
+switch the mode to `restrict`.
+
+Once configured, Magento can enforce policies like these:
+
+*  Any resource, such as `.js`, `.css`, `.jpg`, or `.ttf` files, can only be loaded from the store's domain
+*  Iframes can only include pages from the store itself
+*  AJAX requests can only be sent to the store
+*  Forms can only be sent to the store
+*  Only whitelisted inline scripts and styles to be compiled by browser
```
Comment (jeff-matthews) [inline src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md:46]:
```suggestion
*  Only whitelisted inline scripts and styles can be compiled by browser
```

File: src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+For more details check the `Magento/Csp/etc/config.xml` file. Some domains have already been
```
Comment (jeff-matthews) [inline src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md:48]:
```suggestion
For more details, check the `Magento/Csp/etc/config.xml` file. Some domains have already been
```

File: src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+whitelistedfor modules that require it. For instance if the `Magento_Paypal` module is installed,
```
Comment (jeff-matthews) [inline src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md:49]:
```suggestion
whitelisted for modules that require it. For examplle, if the `Magento_Paypal` module is installed,
```

File: src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+`www.paypal.com` is already whitelisted for the `script-src` policy. Inline scripts as well as styles have also been whitelisted,
```
Comment (jeff-matthews) [inline src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md:50]:
```suggestion
`www.paypal.com` is already whitelisted for the `script-src` policy. If inline scripts and styles have also been whitelisted,
```

File: src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+their hashes will be added to `Content-Security-Policy` header only when inline scripts/styles are not allowed.
```
Comment (jeff-matthews) [inline src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md:51]:
```suggestion
their hashes will be added to the `Content-Security-Policy` header only when inline scripts/styles are not allowed.
```

File: src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+                <value id="devdocs" type="host">https://devdocs.magento.com</value>
+            </values>
+        </policy>
+    </policies>
+</csp_whitelist>
+```
+
+### Whitelist an inline script or style
+
+For stores that have `unsafe-inline` disabled for `style-src` and `script-src` (default for Magento 2.4) inline scripts and styles
```
Comment (jeff-matthews) [inline src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md:138]:
```suggestion
Stores that have `unsafe-inline` disabled for `style-src` and `script-src` (default for Magento 2.4) inline scripts and styles
```

File: src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+must be whitelisted.
+
+You would need to use `Magento\Framework\View\Helper\SecureHtmlRenderer` that is available
```
Comment (jeff-matthews) [inline src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md:141]:
```suggestion
You must use `Magento\Framework\View\Helper\SecureHtmlRenderer`, which is available
```

File: src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md [hunk 8]
```diff
@@ -0,0 +1,282 @@
...
+as `$secureRenderer` variable in _.phtml_ templates to achieve this.
```
Comment (jeff-matthews) [inline src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md:142]:
```suggestion
as a `$secureRenderer` variable in the _.phtml_ templates to achieve this.
```

File: src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md [hunk 9]
```diff
@@ -0,0 +1,282 @@
...
+ 
+ 
+* For `script` or `style` tags containing inline JavaScript/CSS use `renderTag`:
```
Comment (jeff-matthews) [inline src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md:145]:
```suggestion
* For `script` or `style` tags containing inline JavaScript/CSS, use `renderTag`:
```

File: src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md [hunk 10]
```diff
@@ -0,0 +1,282 @@
...
+function someMethod() {
+   ....
+   
+   $html .= $this->secureRenderer->renderTag('style', [], "#element { color: blue } ", false);
+   
+   ....
+}
+```
+ 
+* For event handlers previously defined in HTML attributes use `renderEventListenerAsTag`
```
Comment (jeff-matthews) [inline src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md:169]:
```suggestion
* For event handlers previously defined in HTML attributes, use `renderEventListenerAsTag`:
```

File: src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md [hunk 11]
```diff
@@ -0,0 +1,282 @@
...
+ 
+_Inside a .phtml template_
+```html
+<!-- <div onclick="alert('Old way!')">Old way</div> -->
+
+<div id="alert-div">New way</div>
+<?= $secureRenderer->renderEventListenerAsTag('onclick', 'alert("New way!");', '#alert-div'); ?>
+```
+ 
+* For inline styles previously defined in `style` attribute use `renderStyleAsTag`
```
Comment (jeff-matthews) [inline src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md:179]:
```suggestion
* For inline styles previously defined in `style` attributes, use `renderStyleAsTag`:
```

File: src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md [hunk 12]
```diff
@@ -0,0 +1,282 @@
...
+ 
+_Inside a .phtml template_
+```html
+<!-- <div style="color:blue">Old way</div> -->
+
+<div id="blue-div">New way</div>
+<?= $secureRenderer->renderStyleAsTag('color: blue', '#blue-div'); ?>
+```
+ 
+_Please note that using inline scripts and styles is discouraged in favor of UI Components and classes_
```
Comment (jeff-matthews) [inline src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md:189]:
```suggestion
_Using inline scripts and styles is discouraged in favor of UI Components and classes_
```

**Comment by jeff-matthews** [comment]
Also, @AlexMaxHorkun, please remove the `2.4` from the end of the file name. Change the file name to:

```
src/guides/v2.4/extension-dev-guide/security/content-security-policies.md
```

File: src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md (hunk 3 above)
Comment (AlexMaxHorkun) [reply src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md:49]: double L in example

**Comment by AlexMaxHorkun** [comment]
> Also, @AlexMaxHorkun, please remove the `2.4` from the end of the file name. Change the file name to:
> 
> ```
> src/guides/v2.4/extension-dev-guide/security/content-security-policies.md
> ```

I am only using GitHub to work with this repo so that would be difficult to, can you please do it?

**Comment by jeff-matthews** [comment]
@AlexMaxHorkun, sure, no problem.

File: src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md (hunk 3 above)
Comment (jeff-matthews) [reply src/guides/v2.4/extension-dev-guide/security/content-security-policies2.4.md:49]: oops. i'll fix it when i change the file name. you can disregard this suggestion.

**Comment by jeff-matthews** [comment]
I'll fix the linting errors.

**Comment by jeff-matthews** [comment]
running tests

**Pull request merged**